src/processing/aggregate_counts.py

```diff
@@ -0,0 +1,29 @@
+
+from pyspark.sql import functions as F
+
+def get_aggregate_counts(df, counting_col, alias_name):
+    '''
+    Function to get counts in dataframe 
+    '''
+    df = (df
+        .agg(
+            F.countDistinct(counting_col).alias(alias_name)
+        )
+    )
```

Comment:
df is a poor arguement name, consider something more descriptive, e.g. input_df or {name of dataset}_df

Reply:
it can help to be descriptive, it's "df_unaggregated", and the output, is "df_aggregated"

Then the code for this function would change to:
`from pyspark.sql import functions as F

def get_aggregate_counts(df_unaggregated, counting_col, alias_name):
    '''
    Function to get counts in dataframe 
    '''
    df_aggregated = (df_unaggregated
        .agg(
            F.countDistinct(counting_col).alias(alias_name)
        )
    )
   return df_aggregated
`

Reply:
Yeah, it is more of a linting thing and as this is an example pipeline, future users might not realise that it is locally scoped and start using df in a more open scope. It is a tiny thing though so feel free to keep as is 😄

Reply:
Gotcha, ok cool, I will update!